launcher/teeserver/tee_server.go

```diff
@@ -77,65 +72,65 @@ func (a *attestHandler) getToken(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "text/html")
 
 	switch r.Method {
-	case "GET":
+	case http.MethodGet:
 		// this could call Attest(ctx) directly later.
 		data, err := os.ReadFile(a.defaultTokenFile)
 
 		if err != nil {
-			a.logger.Error(err.Error())
-			w.WriteHeader(http.StatusNotFound)
-			w.Write([]byte("failed to get the token"))
+			err = fmt.Errorf("failed to get the token: %w", err)
+			a.logAndWriteHTTPError(w, http.StatusNotFound, err)
 			return
 		}
 		w.WriteHeader(http.StatusOK)
 		w.Write(data)
 		return
-	case "POST":
-		var tokenReq customTokenRequest
+	case http.MethodPost:
+		var tokenOptions models.TokenOptions
 		decoder := json.NewDecoder(r.Body)
 		decoder.DisallowUnknownFields()
 
-		err := decoder.Decode(&tokenReq)
+		err := decoder.Decode(&tokenOptions)
 		if err != nil {
-			a.logger.Error(err.Error())
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte(err.Error()))
+			a.logAndWriteHTTPError(w, http.StatusBadRequest, err)
```

Comment:
You should prefix this error with some more context like `fmt.Errorf("failed to parse POST body as TokenOptions: %v", err)`. Otherwise the resulting error is not very understandable.

Reply:
Good catch